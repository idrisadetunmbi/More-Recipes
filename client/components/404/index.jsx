import React from 'react';

const styles = {
  textAlign: 'center',
  marginTop: '15rem',
  p: {
    fontFamily: 'Open Sans',
  },
};

export default () => (
  <div style={styles} >
    <h1>404</h1>
    <p style={styles.p}>This page does not exist</p>
  </div>
);
