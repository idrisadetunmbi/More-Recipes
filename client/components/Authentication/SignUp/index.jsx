import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import validator from 'validator';

import { userAuthRequest } from '../../../actions/user';

class SignUp extends React.Component {
  state = {
    username: '',
    email: '',
    password: '',
    passwordConfirm: '',
    fieldError: {
      username: '',
      email: '',
      password: '',
      passwordConfirm: '',
    },
  }

  // eslint-disable-next-line
  onChange = (e) => {
    const inputFieldName = e.target.name;
    let inputFieldValue;

    switch (inputFieldName) {
      case 'username':
        inputFieldValue = e.target.value.trim().toLowerCase();
        this.setState({ [inputFieldName]: inputFieldValue });
        if (
          !validator.isLength(inputFieldValue, { min: 5, max: 15 }) ||
          !validator.isAlphanumeric(inputFieldValue)
        ) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              username: 'username must have at least five alphanumeric characters',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              username: '',
            },
          });
        }
        break;
      case 'password':
        inputFieldValue = e.target.value;
        this.setState({ [inputFieldName]: inputFieldValue });
        if (!validator.isLength(inputFieldValue, { min: 6, max: 25 })) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              password: 'password field requires at least six characters',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              password: '',
            },
          });
        }
        break;
      case 'email':
        inputFieldValue = e.target.value;
        this.setState({ [inputFieldName]: inputFieldValue });
        if (!validator.isEmail(inputFieldValue)) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              email: 'please include a valid email',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              email: '',
            },
          });
        }
        break;
      case 'passwordConfirm':
        inputFieldValue = e.target.value;
        this.setState({ [inputFieldName]: inputFieldValue });
        if (inputFieldValue !== this.state.password) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              passwordConfirm: 'passwords do not match',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              passwordConfirm: '',
            },
          });
        }
        break;
      default:
        break;
    }
  }

  onSubmit = (e) => {
    e.preventDefault();
    if (
      !(Object.values(this.state.fieldError).every(val => val.length === 0))
    ) {
      return;
    }
    const userData = {
      username: this.state.username,
      password: this.state.password,
      email: this.state.email,
      firstName: 'firstname',
      lastName: 'lastname',
    };
    this.props.userAuthRequest(userData, 'signup');
  }

  // eslint-disable-next-line
  componentWillReceiveProps(nextProps) {
    const { authError, authRequestInitiated } = nextProps.user;
    if (authRequestInitiated) {
      return;
    }
    if (authError) {
      const { error } = authError;
      // eslint-disable-next-line
      error.includes('username') ? 
        this.setState({ fieldError: { ...this.state.fieldError, username: 'username has been taken' } }) :
        this.setState({ fieldError: { ...this.state.fieldError, email: 'this email is already registered' } })
    } else {
      this.props.history.replace(this.props.location.state.previousLocation);
    }
  }

  render() {
    const {
      username, email, password, passwordConfirm, fieldError,
    } = this.state;

    return (
      <form onSubmit={this.onSubmit}>
        <div style={{ marginTop: '3rem', marginBottom: '1rem' }}>
          <h5 style={{ fontFamily: 'Raleway' }}>Create an account</h5>
        </div>

        <div className="input-field" style={{ marginBottom: '20px' }}>
          <label htmlFor="username">Username</label>
          <input value={username} required style={{ marginBottom: '0' }} name="username" id="username" type="text" onChange={this.onChange} />
          {fieldError.username.length > 0 && <span style={{ color: 'red' }}>{fieldError.username}</span>}
        </div>

        <div className="input-field" style={{ marginBottom: '20px' }}>
          <label htmlFor="email">Email</label>
          <input value={email} required style={{ marginBottom: '0' }} name="email" id="email" type="email" onChange={this.onChange} />
          {fieldError.email.length > 0 && <span style={{ color: 'red' }}>{fieldError.email}</span>}
        </div>
      
        <div className="input-field" style={{ marginBottom: '20px' }}>
          <label htmlFor="password">Password</label>
          <input value={password} required style={{ marginBottom: '0' }} name="password" id="password" type="password" onChange={this.onChange} />
          {fieldError.password.length > 0 && <span style={{ color: 'red' }}>{fieldError.password}</span>}
        </div>

        <div className="input-field" style={{ marginBottom: '20px' }}>
          <label htmlFor="passwordConfirm">Confirm Password</label>
          <input value={passwordConfirm} required style={{ marginBottom: '0' }} name="passwordConfirm" id="passwordConfirm" onChange={this.onChange} type="password" />
          {fieldError.passwordConfirm.length > 0 && <span style={{ color: 'red' }}>{fieldError.passwordConfirm}</span>}
        </div>

        <button id="download-button" className="btn-large waves-effect waves-light" style={{ textTransform: 'none', display: 'block', backgroundColor: '#444', width: '100%' }}>
          Sign Up
        </button>
        {this.props.user.authRequestInitiated && <div className="progress"><div className="indeterminate" /></div>}
        <p style={{ textAlign: 'center' }}>Already Have an Account? <Link to={{ pathname: '/signin', state: this.props.location.state }}>Sign In</Link> </p>
      </form>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    userAuthRequest: (authData, authType) => dispatch(userAuthRequest(authData, authType)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);
