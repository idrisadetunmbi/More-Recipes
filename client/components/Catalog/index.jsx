import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import InfiniteScroll from 'react-infinite-scroll-component';
import { Link } from 'react-router-dom';

import Recipe from '../Recipe';
import { LoaderWithComponent } from '../reusables';
import './index.scss';
import { fetchRecipes, searchRecipes } from '../../actions/recipes';

/**
 * @export
 * @class Catalog
 * @extends {Component}
 */
export class Catalog extends Component {
  state = {
    searchTerm: '',
    searchMode: false,
  }

  /**
   * @returns {void}
   * @memberOf Catalog
   */
  componentWillUnmount() {
    // scroll to top of the page incase infinite scroll 'scrollTop button'
    // gets clicked
    window.scrollTo(0, 0);
  }

  /**
   * onClick handler for add recipe button
   * @returns {void}
   * @memberOf Catalog
   */
  addRecipeOnClick = () => {
    const { history } = this.props;
    if (!this.props.user.data.token) {
      history.push('/signin', {
        modal: true,
        previousLocation: history.location.pathname,
      });
      return;
    }
    history.push('/recipes/create', {
      modal: true,
      previousLocation: history.location.pathname,
    });
  }

  /**
   * handles enter key press when search input element is focused
   * @param {object} event - DOM event object
   *
   * @returns {void}
   * @memberOf Catalog
   */
  enterKeyPress = (event) => {
    if (event.keyCode === 13) {
      event.stopPropagation();
      const searchInput = document.querySelector('input');
      const searchTerm = searchInput.value;
      if (!searchTerm) {
        return;
      }
      this.setState({ searchMode: true, searchTerm });
      this.props.searchRecipes(searchTerm);
    }
  }

  /**
   * @param {event} event - DOM event object
   *
   * @returns {void}
   * @memberOf Catalog
   */
  handleClearSearchClick = (event) => {
    const searchInput = document.querySelector('input');
    event.stopPropagation();
    searchInput.value = '';
    searchInput.focus();
    event.target.style.display = 'none';
    this.setState({ searchMode: false });
  }

  /**
   * @param {event} event - DOM event object
   *
   * @returns {void}
   * @memberOf Catalog
   */
  handleSearchInputChange = (event) => {
    const { value } = event.target;
    const clearSearchButton = document.getElementById('clear-search');
    if (!value) {
      this.setState({ searchMode: false });
      clearSearchButton.style.display = 'none';
      return;
    }
    clearSearchButton.style.display = 'inline-block';
  }

  renderLoader = () => (
    <div className="infinite-scroll">
      <div className="preloader-wrapper small active">
        <div className="spinner-layer spinner-green-only">
          <div className="circle-clipper left">
            <div className="circle" />
          </div>
          <div className="gap-patch">
            <div className="circle" />
          </div>
          <div className="circle-clipper right">
            <div className="circle" />
          </div>
        </div>
      </div>
    </div>
  )

  /**
   * @param {Array} recipes - list of recipes to render
   *
   * @returns {JSX.Element} - list of recipe cards
   * @memberOf Catalog
   */
  renderRecipes = recipes =>
    recipes.map(recipe => (
      <Link key={recipe.id} to={`/recipes/${recipe.id}`}>
        <Recipe gridStyle="l4 s12 m6" recipe={recipe} />
      </Link>
    ))

  renderSearch = () =>
    (
      <div id="search-section" className="row">
        <div className="col offset-l3 l9 m12 s12">
          <input
            placeholder="Search Recipe"
            type="search"
            onFocus={() => { document.addEventListener('keydown', this.enterKeyPress); }}
            onBlur={() => { document.removeEventListener('keydown', this.enterKeyPress); }}
            onChange={this.handleSearchInputChange}
          />
          <i
            onClick={this.handleClearSearchClick}
            id="clear-search"
            className="material-icons"
          >
            cancel
          </i>
        </div>
      </div>
    )

  renderSuggestionSection = () => (
    <div className="col l3" id="suggestion-section">
      <h5>Today{'\''}s Combo</h5>
      <div className="divider" />
      <div className="recipe">
        <div className="card">
          <div className="card-image">
            <img alt="" src="images/sample_recipes/recipe0116-xl-farro-with-vinegar-glazed-sweet-potato-and-apples.jpg" />
          </div>
          <div className="card-content">
            <span className="card-title">Farrow with vinegar, glazed sweet potato and apples</span>
            <p>I am a very simple card. I am good at containing small&nbsp;
          bits of information. I am convenient because I require&nbsp;
          little markup to use effectively.
            </p>
          </div>
          <div className="card-action">
            <a href="#">This is a link</a>
          </div>
        </div>
      </div>
    </div>
  )

  renderAddRecipeButton = () => (
    <div className="fixed-action-btn">
      <a className="btn-floating btn-large pulse" onClick={this.addRecipeOnClick}>
        <i className="material-icons">add</i>
      </a>
    </div>
  )

  renderCatalogWithInfiniteScroll = () => {
    const scrollEndOnClick = (event) => {
      event.preventDefault();
      $('html, body').animate({
        scrollTop: $('#catalog-component').offset().top,
      }, 500);
    };
    const endMessage = () => (
      <h5 className="infinite-scroll">
        There are no more recipes.
        <a onClick={scrollEndOnClick}> Go to Top</a>
      </h5>
    );

    const { recipes } = this.props;
    return (
      <LoaderWithComponent
        showLoader={recipes.requestInitiated && !recipes.recipes.length}
        component={
          <InfiniteScroll
            next={this.props.fetchRecipes}
            hasMore={!recipes.fetchedAll}
            loader={this.renderLoader()}
            scrollThreshold={0.8}
            endMessage={endMessage()}
            style={{ position: 'relative' }}
          >
            {this.renderRecipes(recipes.recipes)}
          </InfiniteScroll>}
      />
    );
  }

  renderSearchResults = () => {
    const { searchTerm } = this.state;
    const searchResults = this.props.recipes
      .searchResults[searchTerm.toLowerCase()];
    const endMessage = <h5 className="infinite-scroll">There are no more recipes matching the search term</h5>;

    return (
      <LoaderWithComponent
        showLoader={!searchResults}
        component={
          searchResults && searchResults.results.length ?
            <InfiniteScroll
              next={() => this.props.searchRecipes(this.state.searchTerm)}
              hasMore={!searchResults.matchedAll}
              loader={this.renderLoader()}
              scrollThreshold={0.8}
              endMessage={endMessage}
              style={{ position: 'relative' }}
            >
              {
                this.renderRecipes(this.props.recipes.recipes.filter(recipe =>
                  searchResults.results.includes(recipe.id)))
              }
            </InfiniteScroll> :
            <h5>No results found for {this.state.searchTerm}</h5>
        }
      />
    );
  }

  /**
   * @returns {JSX.Element} JSX element representing the whole catalog page
   * @memberOf Catalog
   */
  render() {
    return (
      <div id="catalog-component">
        {this.renderSearch()}
        <div id="main-divider" className="divider" />
        <div className="row">
          {this.renderSuggestionSection()}
          <div className="col l9 s12">
            <h5>{this.state.searchMode ?
              `Search Results for ${this.state.searchTerm}` : 'Featured Recipes'}
            </h5>
            <div className="divider" id="gallery-before" />
            {
              this.state.searchMode ?
                this.renderSearchResults() :
                this.renderCatalogWithInfiniteScroll()
            }
          </div>
        </div>
        {this.renderAddRecipeButton()}
      </div>
    );
  }
}

Catalog.propTypes = {
  recipes: PropTypes.shape({
    recipes: PropTypes.array,
    fetchedAll: PropTypes.bool,
    requestInitiated: PropTypes.bool.isRequired,
    searchResults: PropTypes.shape().isRequired,
  }).isRequired,

  history: PropTypes.shape({
    location: PropTypes.shape({
      pathname: PropTypes.string,
    }),
  }).isRequired,

  user: PropTypes.shape({
    data: PropTypes.shape({
      token: PropTypes.string,
    }),
  }).isRequired,

  fetchRecipes: PropTypes.func.isRequired,
  searchRecipes: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
  recipes: state.recipes,
});

const mapDispatchToProps = dispatch => ({
  fetchRecipes: () => dispatch(fetchRecipes()),
  searchRecipes: searchTerm => dispatch(searchRecipes(searchTerm)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Catalog);
