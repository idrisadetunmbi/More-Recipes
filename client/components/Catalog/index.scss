#catalog-component {
  #main-divider {
    margin-left: 3em;
    margin-right: 3em; 
    margin-top: 0; 
    display: block;
  }
  
  #search-section {
    margin: 3em;
    margin-bottom: 0;
    input {
      margin-left: -1em;
      border: 1px solid #9e9e9e;
      padding-left: .8em;
      font-size: 1em;
      font-family: 'Lato';
      border-radius: .25rem;
    };
    div {
      position: relative;
    }
    i {
      position: absolute;
      top: 0.7rem;
      right: 1.5rem;
      cursor: pointer;
      display: none;
    }
  };

  #suggestion-section {
    padding: 3em;
    margin-top: 1em;
    .divider {
      margin-bottom: 2em;
    }
  };

  h5 {
    font-family: 'Raleway';
    font-style: italic;
    font-size: 1.5rem;
    padding-left: .5rem;
    text-align: center;
  };

  #gallery-before {
    margin: 0 .7rem 1rem .5rem;
  }

  .infinite-scroll {
    position: absolute;
    bottom: -3rem;
    margin: 0 auto;
    text-align: center;
    left: 0;
    right: 0;    
  }
}