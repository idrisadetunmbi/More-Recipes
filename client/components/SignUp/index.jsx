import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import validator from 'validator';
import PropTypes from 'prop-types';
import { InputField } from '../reusables';

import { userAuthRequest } from '../../actions/user';
import { showToast } from '../../utils';
import './index.scss';

/**
 *
 *
 * @class SignUp
 * @extends {Component}
 */
export class SignUp extends Component {
  state = {
    username: '',
    email: '',
    password: '',
    passwordConfirm: '',
    fieldError: {
      username: null,
      email: null,
      password: null,
      passwordConfirm: null,
    },
  }

  /**
   * @param {Object} nextProps
   *
   * @returns {void}
   * @memberOf SignUp
   */
  componentWillReceiveProps(nextProps) {
    const { userRequestInitiated, userRequestError } = nextProps.user;
    if (userRequestInitiated) {
      return;
    }
    if (userRequestError) {
      const { error } = userRequestError;
      // eslint-disable-next-line
      error.includes('username') ?
        this.setState({
          fieldError: {
            ...this.state.fieldError, username: 'username has been taken',
          },
        }) :
        this.setState({
          fieldError: {
            ...this.state.fieldError, email: 'this email is already registered',
          },
        });
    } else {
      showToast('You have signed up successfully!!!');
      this.props.history.replace(this.props.location.state.previousLocation);
    }
  }

  /**
   * @param {Object} event
   *
   * @returns {void}
   * @memberOf SignUp
   */
  onChange = (event) => {
    const inputFieldName = event.target.name;
    let inputFieldValue;

    switch (inputFieldName) {
      case 'username':
        inputFieldValue = event.target.value.trim().toLowerCase();
        this.setState({ [inputFieldName]: inputFieldValue });
        if (
          !validator.isLength(inputFieldValue, { min: 5, max: 15 }) ||
          !validator.isAlphanumeric(inputFieldValue)
        ) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              username: 'username must have at least five alphanumeric characters',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              username: '',
            },
          });
        }
        break;
      case 'password':
        inputFieldValue = event.target.value;
        this.setState({ [inputFieldName]: inputFieldValue });
        if (!validator.isLength(inputFieldValue, { min: 6, max: 25 })) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              password: 'password field requires at least six characters',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              password: '',
            },
          });
        }
        break;
      case 'email':
        inputFieldValue = event.target.value;
        this.setState({ [inputFieldName]: inputFieldValue });
        if (!validator.isEmail(inputFieldValue)) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              email: 'please include a valid email',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              email: '',
            },
          });
        }
        break;
      case 'passwordConfirm':
        inputFieldValue = event.target.value;
        this.setState({ [inputFieldName]: inputFieldValue });
        if (inputFieldValue !== this.state.password) {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              passwordConfirm: 'passwords do not match',
            },
          });
        } else {
          this.setState({
            fieldError: {
              ...this.state.fieldError,
              passwordConfirm: '',
            },
          });
        }
        break;
      default:
        break;
    }
  }

  /**
   * @param {Object} event
   *
   * @returns {void}
   * @memberOf SignUp
   */
  onSubmit = (event) => {
    event.preventDefault();
    if (
      !(Object.values(this.state.fieldError).every(val => val.length === 0))
    ) {
      return;
    }
    const userData = {
      username: this.state.username,
      password: this.state.password,
      email: this.state.email,
    };
    this.props.signUpUser(userData);
  }

  /**
   *
   * @returns {Object} sign up form element
   * @memberOf SignUp
   */
  render() {
    const {
      username, email, password, passwordConfirm, fieldError,
    } = this.state;

    return (
      <form onSubmit={this.onSubmit} className="auth-component">
        <div style={{ marginTop: '3rem', marginBottom: '1rem' }}>
          <h5>Create an account</h5>
        </div>

        <InputField
          label="username"
          value={username}
          type="text"
          onChange={this.onChange}
          required
          fieldError={fieldError.username}
        />

        <InputField
          label="email"
          value={email}
          required
          type="email"
          onChange={this.onChange}
          fieldError={fieldError.email}
        />

        <InputField
          label="password"
          value={password}
          required
          type="password"
          onChange={this.onChange}
          fieldError={fieldError.password}
        />

        <InputField
          label="passwordConfirm"
          value={passwordConfirm}
          required
          type="password"
          onChange={this.onChange}
          fieldError={fieldError.passwordConfirm}
        />

        <button id="form-submit" className="btn-large waves-effect waves-light">
          Sign Up
        </button>
        {this.props.user.userRequestInitiated && <div className="progress"><div className="indeterminate" /></div>}
        <p>Already Have an Account? <Link to={{ pathname: '/signin', state: this.props.location.state }}>Sign In</Link> </p>
      </form>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
});

const mapDispatchToProps = dispatch => ({
  signUpUser: userData =>
    dispatch(userAuthRequest(userData, 'signup')),
});

SignUp.propTypes = {
  user: PropTypes.shape({
    userRequestInitiated: PropTypes.bool,
    userRequestError: PropTypes.object,
  }).isRequired,
  history: PropTypes.shape({
    replace: PropTypes.func,
  }).isRequired,
  location: PropTypes.shape().isRequired,
  signUpUser: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);
