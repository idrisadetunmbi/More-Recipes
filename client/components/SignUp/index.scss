.auth-component {
  h5 {
    font-family: 'Raleway'
  }

  .input-field {
    margin-bottom: 20px;
    input {
      margin-bottom: 0;      
    }
    span {
      color: red;
    }
  }

  #form-submit {
    text-transform: none;
    display: block;
    background-color: #444;
    width: 100%;
  }

  p {
    text-align: center;
  }
}