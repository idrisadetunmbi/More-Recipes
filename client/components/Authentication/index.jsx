import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import SignIn from './SignIn';
import SignUp from './SignUp';

export default class Authentication extends React.Component {

  componentDidMount() {
    document.addEventListener('keydown', this.escKeyPress);
    document.body.style.overflow = 'hidden';
  }

  componentWillUnmount() {
    document.body.style.overflow = '';
    document.removeEventListener('keydown', this.escKeyPress);
  }
  
  back = (e) => {
    e.stopPropagation();
    this.props.history.replace(this.props.location.state.previousLocation);
  };

  escKeyPress = (e) => {
    if (e.keyCode === 27) {
      this.back(e);
    }
  };

  render() {
    return (
      <div onClick={this.back}>
        <div className="modal-overlay" style={{ zIndex: 1002, display: 'block', opacity: 0.5 }} />
        <div
          className="modal open"
          style={{
            zIndex: 1003,
            display: 'block',
            opacity: 1,
            transform: 'scaleX(1)',
            top: '10%',
            maxHeight: '100% important',
        }}
        >
          <div className="modal-content" onClick={(e) => e.stopPropagation()}>
            <div className="container">
              <div className="row">
                <div className="">
                  <Switch>
                    <Route path="/signin" component={SignIn} />
                    <Route path="/signup" component={SignUp} />
                  </Switch>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
