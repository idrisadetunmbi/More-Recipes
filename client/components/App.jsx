import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Route, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import LandingPage from './LandingPage';
import RecipeDetails from './RecipeDetails';
import NavBar from './NavBar';
import Modal from './Modal';
import Catalog from './Catalog';
import UserProfile from './UserProfile';
import NoMatch from './404';

import { fetchRecipes } from '../actions/recipes';

/**
 *
 *
 * @class App
 * @extends {Component}
 */
export class App extends Component {
  // eslint-disable-next-line
  previousLocation = this.props.location

  /**
   * @returns {void}
   * @memberOf App
   */
  componentWillMount() {
    this.props.fetchRecipes();
  }

  /**
   * @param {any} nextProps
   *
   * @returns {void}
   * @memberOf App
   */
  componentWillUpdate(nextProps) {
    localStorage.setItem('user', JSON.stringify(nextProps.userData));
    const { location } = this.props;
    if (
      nextProps.history.action !== 'POP' &&
      (!location.state || !location.state.modal)
    ) {
      this.previousLocation = this.props.location;
    }
  }

  /**
   * @returns {void}
   *
   * @memberOf App
   */
  render() {
    const { location } = this.props;
    const isModal = !!(
      location.state &&
      location.state.modal &&
      this.previousLocation !== location
    );

    return (
      <div>
        <Route component={NavBar} />
        <Switch location={isModal ? this.previousLocation : location}>
          <Route
            exact
            path="/"
            render={() => (
              this.props.userData.token ? (
                <Redirect to="/catalog" />) :
                (<LandingPage />)
            )}
          />
          <Route path="/catalog" component={Catalog} />
          <Route path="/recipes/:recipeId" component={RecipeDetails} />
          <Route path="/user" component={UserProfile} />
          <Route component={NoMatch} />
        </Switch>
        {isModal ? <Route component={Modal} /> : null }
      </div>
    );
  }
}

App.propTypes = {
  fetchRecipes: PropTypes.func.isRequired,
  userData: PropTypes.shape({
    token: PropTypes.string,
  }).isRequired,
};

export const mapStateToProps = state => ({
  userData: state.user.data,
});

export const mapDispatchToProps = dispatch => ({
  fetchRecipes: () => dispatch(fetchRecipes()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
