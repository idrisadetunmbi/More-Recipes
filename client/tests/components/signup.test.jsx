import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter as Router } from 'react-router-dom';

import { SignUp } from '../../components/SignUp';

describe('SignUp component', () => {
  test('matches snapshot', () => {
    const props = { user: {}, location: {} };
    const component = renderer.create(<Router><SignUp {...props} /></Router>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
