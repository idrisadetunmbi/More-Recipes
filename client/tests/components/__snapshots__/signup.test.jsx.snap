// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component matches snapshot 1`] = `
<form
  className="auth-component"
  onSubmit={[Function]}
>
  <div
    style={
      Object {
        "marginBottom": "1rem",
        "marginTop": "3rem",
      }
    }
  >
    <h5>
      Create an account
    </h5>
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="username"
    >
      username
    </label>
    <input
      id="username"
      name="username"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="email"
    >
      email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="password"
    >
      password
    </label>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="passwordConfirm"
    >
      passwordConfirm
    </label>
    <input
      id="passwordConfirm"
      name="passwordConfirm"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
  </div>
  <button
    className="btn-large waves-effect waves-light"
    id="form-submit"
  >
    Sign Up
  </button>
  <p>
    Already Have an Account? 
    <a
      href="/signin"
      onClick={[Function]}
    >
      Sign In
    </a>
     
  </p>
</form>
`;
